prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -493,13 +495,34 @@ ssize_t rxr_pkt_post_read_to_copy_data(struct rxr_ep *rxr_ep,
 	assert(iov_idx >=0 && iov_idx < rx_entry->iov_count);
 	assert(efa_ep_is_cuda_mr(rx_entry->desc[iov_idx]));
 	assert(iov_offset + data_size <= rx_entry->iov[iov_idx].iov_len);
+	assert(!pkt_entry->next);
 
 	dest_buf = (char *)rx_entry->iov[iov_idx].iov_base + iov_offset;
 
-	pkt_entry->state = RXR_PKT_ENTRY_COPY_BY_READ;
-	assert(pkt_entry->mr);
+	if (!pkt_entry->mr) {
+		assert(pkt_entry->type == RXR_PKT_ENTRY_OOO ||
+		       pkt_entry->type == RXR_PKT_ENTRY_UNEXP);
+		/* ooo and unexp packet entry's memory is not registered with device,
+		 * so we allocate a packet entry from rx_readcopy_pkt_pool (which is registered),
+		 * and copy data to the new packet entry
+		 */
+		pkt_offset = data - (char *)pkt_entry->pkt;
+		assert(pkt_offset > sizeof(struct rxr_base_hdr));
+
+		pkt_entry_copy = rxr_pkt_entry_clone(rxr_ep, rxr_ep->rx_readcopy_pkt_pool,
```

Comment:
On second thought after our conversation `rxr_ep->tx_pkt_efa_pool` can be used here instead of creating another pool.

Reply:
Update the first commit to use ```rxr_ep->tx_pkt_efa_pool``` for this read request.